database-commons/src/main/java/io/cdap/plugin/db/source/AbstractDBSource.java

```diff
@@ -114,7 +119,7 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
 
     if (sourceConfig.canConnect()) {
       try {
-        stageConfigurer.setOutputSchema(getSchema(driverClass));
+        stageConfigurer.setOutputSchema(getSchema(driverClass, collector));
```

Comment:
we are already adding the exception in failure collector below by catching it, why is this extra step needed?

Reply:
also can you please add sql state and error code in the message on line 127?

Reply:
Added SQL error details.